pkg/lieutenant/lieutenant.go

```diff
@@ -0,0 +1,64 @@
+package lieutenant
+
+import (
+	"context"
+	"fmt"
+
+	lieutenantv1alpha1 "github.com/projectsyn/lieutenant-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Client interface {
+	GetClusterFacts(context.Context, string) (map[string]string, error)
+}
+
+type client_impl struct {
+	Client    client.Client
+	Namespace string
+}
+
+type Config struct {
+	Host      string
+	Token     string
+	Namespace string
+}
+
+func NewLieutenantClient(config Config) (Client, error) {
```

Comment:
It's not necessarily the responsibility of this package to define the interface of Client. The package accepting the interface should specify it.

This would also solve the awkward `client_impl` naming issue.